File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/adapter/AllCurrenciesAdapter.java
package io.github.skywalkerdarren.simpleaccounting.adapter;

import android.content.Context;

import io.github.skywalkerdarren.simpleaccounting.R;
import io.github.skywalkerdarren.simpleaccounting.base.BaseDataBindingAdapter;
import io.github.skywalkerdarren.simpleaccounting.databinding.ItemCurrencyBinding;
import io.github.skywalkerdarren.simpleaccounting.model.AppRepository;
import io.github.skywalkerdarren.simpleaccounting.model.entity.Currency;
import io.github.skywalkerdarren.simpleaccounting.model.entity.CurrencyInfo;
import io.github.skywalkerdarren.simpleaccounting.util.AppExecutors;
import kotlin.Pair;

/**
 * @author darren
 * @date 2018/4/13
 */

public class AllCurrenciesAdapter extends BaseDataBindingAdapter<Pair<Currency, CurrencyInfo>, ItemCurrencyBinding> {
    private final AppRepository mRepository;

    public AllCurrenciesAdapter(Context context) {
        super(R.layout.item_currency, null);
        mRepository = AppRepository.getInstance(new AppExecutors(), context);
    }

    @Override
    protected void convert(ItemCurrencyBinding binding, Pair<Currency, CurrencyInfo> item) {
        binding.setCurrency(item.getFirst());
        binding.setInfo(item.getSecond());
        binding.favCurrency.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (item.getFirst() != null) {
                mRepository.setCurrencyFav(item.getFirst().getName(), isChecked);
            }
        });
    }
}

File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/adapter/AllCurrenciesAdapter.kt
package io.github.skywalkerdarren.simpleaccounting.adapter

import com.chad.library.adapter.base.viewholder.BaseDataBindingHolder
import io.github.skywalkerdarren.simpleaccounting.R
import io.github.skywalkerdarren.simpleaccounting.base.BaseDataBindingAdapter
import io.github.skywalkerdarren.simpleaccounting.databinding.ItemCurrencyMultiBinding
import io.github.skywalkerdarren.simpleaccounting.model.entity.CurrencyAndInfo
import io.github.skywalkerdarren.simpleaccounting.view_model.CurrencyFavViewModel

/**
 * @author darren
 * @date 2018/4/13
 */
class AllCurrenciesAdapter(private val viewModel: CurrencyFavViewModel) :
        BaseDataBindingAdapter<CurrencyAndInfo, ItemCurrencyMultiBinding>
        (R.layout.item_currency_multi, null) {
    override fun convert(holder: BaseDataBindingHolder<ItemCurrencyMultiBinding>, item: CurrencyAndInfo) {
        holder.dataBinding?.currency = item.currency
        holder.dataBinding?.info = item.currencyInfo
        holder.dataBinding?.favCurrency?.setOnClickListener {
            viewModel.setCurrencyFav(item.currency, holder.dataBinding?.favCurrency?.isChecked
                    ?: false)
        }
    }
}